src/couch_replicator_scheduler.erl

```diff
@@ -30,23 +30,25 @@
 -export([handle_config_change/5, handle_config_terminate/3]).
 
 %% types
--type event_type() :: started | stopped | crashed.
+-type event_type() :: started | stopped | {crashed, any()}.
 -type event() :: {Type:: event_type(), When :: erlang:timestamp()}.
 -type history() :: [Events :: event()].
 
 %% definitions
 -define(MAX_HISTORY, 20).
--define(MINIMUM_CRASH_INTERVAL, 60 * 1000000).
 
 -define(DEFAULT_MAX_JOBS, 100).
 -define(DEFAULT_MAX_CHURN, 20).
 -define(DEFAULT_SCHEDULER_INTERVAL, 60000).
+-define(ERROR_BACKOFF_MILLIS, 30000).
 -record(state, {interval, timer, max_jobs, max_churn}).
 -record(job, {
           id :: job_id(),
           rep :: #rep{},
           pid :: pid(),
-          history :: history()}).
+          history :: history(),
+          errors :: integer(),
+          errored_at :: integer()}).
```

Comment:
Is this an integer or a float? Initialized below to 0.0 but not sure if it we still mean to be float.